docs/UsingSpiffdemo/Basic_Tools.md

```diff
@@ -215,12 +207,70 @@ during the process execution.
 
 
 ## Writing Scripts 
+Writing scripts refers to the process of creating custom code or scripts to increase the functionality and automation of a software application or system. 
+
+In SpiffArena, writing scripts involves creating code snippets or 
+specific tasks, manipulate data, or handle complex logic.
+
+The scripting language used for writing scripts in SpiffArena is  Python, a widely used programming language. Python provides a rich set of libraries, frameworks, and tools that facilitate script development, making it a popular choice for implementing custom logic and automation.
+
+Let's explore an example of a Script Task in our basics section:
+
+1. **Start Event and User Task - "Form"**
+
+The process begins with a Start Event, followed by a User Task named "Form." Users will fill out the form, and the data will be passed to the next task, which is a Script Task.
+![](Images/Script_Example_S2.png)
+
+2. **Script Task to collect data**
+
+In the Script Task, we have created a script that collects three variables from the form and calculates a score based on certain conditions. The score is then stored in the "score" variable.
+
+![](Images/Script_Example_S3.png)
+
+Here's the script:
+
+![](Images/Script_Example_S4.png)
 
+3. **Manual Task with Pre-Script**
+
+After the Script Task, we have a Manual Task with a pre-script and instructions to display the score. 
+![](Images/Script_Example_S5.png)
+
+- **Prescript** is added as an example, While you can have tasks that are dedicated scripts, it can become a bit noisy, and we want our diagrams to convey a clear sense of the business logic and rules. For this reason it is also possible to add scripts to all Task types - using Pre and Post Scripts. This manual task contains a pre-script that also calculated PI using the Leibniz’s formula.
+
+- **Post Scripts** are also available on most task types, but they execute AFTER the task is completed. These are great for user forms where you want to modify and clean up the form results before moving on to the next task.
 
-asdas
 ## Making Decisions
+DMN tables are powerful tools for modeling and implementing business rules and decision logic. They allow you to define rules and their associated conditions and actions in a structured manner. By evaluating the conditions in each rule, the DMN engine can determine which rules are triggered based on the provided inputs and execute the corresponding actions. This provides a flexible and configurable approach to decision-making in various scenarios.
+
+More detailed information [here](https://spiff-arena.readthedocs.io/en/latest/appendices/bpmn_references.html#).
```

Comment:
instead of the full url, `https://spiff-arena.readthedocs.io/en/latest/appendices/bpmn_references.html`, you can use `/appendices/bpmn_references.md`

Reply:
Basically I did changed it but it gives build error 
![image](https://github.com/sartography/spiff-arena/assets/96994784/5e0e1d55-9d60-4bfd-ab90-0a417cbbb5d2)

Reply:
It should look like this @usama9500:
More detailed information \[here\]\(/appendices/bpmn_references\).

The extension should not be included.

Reply:
More detailed information [here](https://spiff-arena.readthedocs.io/en/latest/appendices/bpmn_references.html#).

The DMN section is no longer under Appendices after the changes merged yesterday they are now under BPMN which Dan requested we put under Building Diagrams. Just worried that your link might not work.